File: aoi0005_lab01/aoi0005_lab01.java
/*ahmed ibrahim 
1001820005
java version "18.0.2.1" 2022-08-18
Windows
*/
import java.io.File; //import to get info abt the file

public class aoi0005_lab01 {
    public static int dirSpace(File directory) 
    {
        int localSum = 0; //holds the total sum of the size of the files
        File[] file = directory.listFiles(); //array that holds all files within a directory
        for (File it : file) //iterator to iterate through all the files
        {
            if (it.getName().equals(".") || it.getName().equals("..")) 
            {
                continue;
            }
            if (it.isFile()) // if it's a file
            {
                localSum += it.length(); //get the size of file and add it to the rest
            } 
            if (it.isDirectory()) //if it's a directory
            {
                localSum += dirSpace(it); //recursively call the function
            }
        }
        return localSum;
    }

    public static void main(String[] args) {
        File targetDir = new File("."); // gets current directory
        System.out.print(dirSpace(targetDir)); // print out size in bytes
    }
}

File: aoi0005_lab01/aoi0005_lab01.py
#Ahmed Ibrahim
#1001820005
#Windows
#Python 3.9.5
import os #import to access info abt OS 
def dirSpace(Directory): #function to get size of all files in directory
    localSum = 0
    for files in os.listdir(Directory): #loops through all the files
        file = os.path.join(Directory, files) 
        if file == "." or file == "..":
            continue
        if os.path.isfile(file): #if it's a file
            localSum += os.path.getsize(file) #get it's size
        if os.path.isdir(file): #if it's a directory
            localSum += dirSpace(file) #recursively call the function to get the size of all files in a directory
    return localSum
targetDir = "."
print(dirSpace(targetDir))